### PR TITLE
added apiview for sale reciept

### PR DESCRIPTION
**Pull request by pavanpulicherla18345**

### CODE REVIEW

File: apps/sales/views.py [hunk 1]
```diff
@@ -1457,6 +1457,428 @@ def put(self, request, pk, *args, **kwargs):
...
+            sale_receipt = get_object_or_404(SaleReceipt, pk=pk)
+            sale_receipt_serializer = SaleReceiptSerializer(sale_receipt)
+
+            # # Retrieve assignment_history_data
+            # assignment_history_data = self.get_related_data(LeadAssignmentHistory, LeadAssignmentHistorySerializer, 'sale_receipt_id', pk)
+            # assignment_history_data = assignment_history_data if assignment_history_data else []
+
+            # # Retrieve interaction_data
+            # interaction_data = self.get_related_data(LeadInteractions, LeadInteractionsSerializer, 'sale_receipt_id', pk)
+            # interaction_data = interaction_data if interaction_data else []
```
Comment (QodeFarm) [inline apps/sales/views.py:1507]: remove above commented code.

File: apps/sales/views.py [hunk 2]
```diff
@@ -1457,6 +1457,428 @@ def put(self, request, pk, *args, **kwargs):
...
+
+
+        This "create_new_history" function creates the new instance in 'LeadAssignmentHistory' model
+        with required fields to be updated.
+
+        def create_new_history(self,update_fields):
+            assignment_history = {}
+            assignments_history = generic_data_creation(self, [assignment_history], LeadAssignmentHistorySerializer, update_fields)
+            assignments_history = assignments_history if assignments_history else []
+            logger.info('LeadAssignmentHistory - created*') '''
```
Comment (QodeFarm) [inline apps/sales/views.py:1673]: What is above commented? It is not specific to sale_receipt

File: apps/sales/views.py [hunk 3]
```diff
@@ -1457,6 +1457,428 @@ def put(self, request, pk, *args, **kwargs):
...
+
+        if sale_receipt_data:
+            '''
+            This block performs updation of end date in case of 'assignee_id' and 'sale_receipt_id' is changed.
+            (detects changes in 'SaleReceipt' payload)
+            Adds new record in 'LeadAssignmentHistory' if 'assignee_id' in 'SaleReceipt' is changed
+            by leaving end date as null.
+            Updates old record in 'LeadAssignmentHistory' if 'sale_receipt_id' in 'SaleReceipt' is changed
+            by adding end date
+            '''
```
Comment (QodeFarm) [inline apps/sales/views.py:1683]: Correct above comments, what is LeadAssignmentHistory here?

File: config/settings.py [hunk 4]
```diff
@@ -132,7 +132,7 @@
 DATABASES = {
     'default': {
         "ENGINE": "django.db.backends.mysql",
-        "NAME": "cnl",
+        "NAME": "purchaseupdation",
```
Comment (QodeFarm) [inline config/settings.py:135]: what is this purchaseupdation? It should be cnl right?

File: apps/sales/views.py (hunk 1 above)
Comment (pavanpulicherla18345) [reply apps/sales/views.py:1507]: Okay Thulasi.

File: apps/sales/views.py (hunk 2 above)
Comment (pavanpulicherla18345) [reply apps/sales/views.py:1673]: yeah Thulasi removed , I pasted sale_reciept code which was sent by Pramod , he mistakenly added it Thulasi.

File: apps/sales/views.py (hunk 3 above)
Comment (pavanpulicherla18345) [reply apps/sales/views.py:1683]: Removed Thulasi

File: config/settings.py (hunk 4 above)
Comment (pavanpulicherla18345) [reply config/settings.py:135]: Sorry Thulasi I gave local database name there.(Now changed to cnl)

**Pull request merged**